### PR TITLE
[6377] Added name and avatar for users

### PR DESCRIPTION
**Pull request by KarolJagodzinski**
**Ticket**: [BAU-6377](https://bugzilla.stxnext.pl/show_bug.cgi?id=6377)

replaced users id to name in dropdown
added images
added parsing xml
added utils and views

### CODE REVIEW

File: src/presence_analyzer/tests.py [hunk 1]
```diff
@@ -52,8 +61,15 @@ def test_api_users(self):
         self.assertEqual(resp.status_code, 200)
         self.assertEqual(resp.content_type, 'application/json')
         data = json.loads(resp.data)
-        self.assertEqual(len(data), 2)
-        self.assertDictEqual(data[0], {'user_id': 10, 'name': 'User 10'})
+        # import ipdb; ipdb.set_trace()
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: :sos:

File: src/presence_analyzer/tests.py [hunk 2]
```diff
@@ -177,6 +214,7 @@ def test_get_data(self):
         Test parsing of CSV file.
         """
         data = utils.get_data()
+        # import ipdb; ipdb.set_trace()
```
Comment (mjelonek92) [inline src/presence_analyzer/tests.py]: :sos:

File: src/presence_analyzer/views.py [hunk 3]
```diff
@@ -46,11 +47,27 @@ def users_view():
...
+    data_xml = get_data_xml()
     return [
-        {'user_id': i, 'name': 'User {0}'.format(str(i))}
-        for i in data.keys()
-    ]
+        {
+            'user_id': i,
+            'name': data_xml[i]['name'],
+            'avatar': data_xml[i]['avatar']
+        } for i in data_xml.keys()]
```
Comment (mjelonek92) [inline src/presence_analyzer/views.py]: No need to call `.keys`. Also please use something more descriptive than `i`.

File: src/presence_analyzer/utils.py [hunk 4]
```diff
@@ -73,6 +74,39 @@ def get_data():
...
 
 
+def get_data_xml():
+    """
+    Get data from xml.
+
+    It creates structure like this:
+    data = {
+        'user_id': {
+                'avatar': 'api/images/users/user_id',
```
Comment (mjelonek92) [inline src/presence_analyzer/utils.py]: Missing host part no? I.e. `schema://server:port/api/...`

File: src/presence_analyzer/script.py [hunk 5]
```diff
@@ -111,3 +111,11 @@ def action_stop(dry_run=False):
         _serve('stop', dry_run=dry_run)
 
     werkzeug.script.run()
+
+
+def download_xml():
+    """Download xml files from stx website."""
```
Comment (dpolcyn) [inline src/presence_analyzer/script.py]:
```
"""
Download xml files from stx website.
"""
```

File: src/presence_analyzer/utils.py [hunk 6]
```diff
@@ -73,6 +74,39 @@ def get_data():
     return data
 
 
+def get_data_xml():
+    """
+    Get data from xml.
+
+    It creates structure like this:
+    data = {
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]:
````
data = {
    'user_id': {
        'avatar': 'scheme://server:port/api/images/users/user_id',
        'name': 'Jan K.',
    }
}
```

File: src/presence_analyzer/utils.py [hunk 7]
```diff
@@ -73,6 +74,39 @@ def get_data():
...
+        'user_id': {
+                'avatar': 'scheme://server:port/api/images/users/user_id',
+                'name': 'Jan K.',
+            }
+        }
+    """
+    tree = etree.parse(app.config['DATA_XML'])
+    root = tree.getroot()
+    children = root.getchildren()
```
Comment (dpolcyn) [inline src/presence_analyzer/utils.py]: ```children``` is bad variable name here. Tree can contains a lot of childrens and thats why you are getting values like this:```children[0]```. Just use ```.find``` to get what you need and parse data.

File: runtime/data/users.xml [hunk 8]
```diff
@@ -0,0 +1,350 @@
...
+            <avatar>/api/images/users/141</avatar>
+            <name>Adam P.</name>
+        </user>
+        <user id="176">
+            <avatar>/api/images/users/176</avatar>
+            <name>Adrian K.</name>
+        </user>
+        <user id="999">
+            <avatar>/api/images/users/176</avatar>
+            <name>ZZZZZZ</name>
```
Comment (dpolcyn) [inline runtime/data/users.xml]: ??? This user realy exist?

File: src/presence_analyzer/script.py [hunk 9]
```diff
@@ -42,7 +42,9 @@ def make_debug(global_conf={}, **conf):
 
 # bin/flask-ctl shell
 def make_shell():
-    """Interactive Flask Shell"""
+    """
```
Comment (mjelonek92) [inline src/presence_analyzer/script.py]: Why change docstrings that were correct?

Comment (KarolJagodzinski) [inline src/presence_analyzer/script.py]: I had to change docstring for my script(5th comment), so I changed for all. That's incorrect?

Comment (mjelonek92) [inline src/presence_analyzer/script.py]: One-line docstrings are correct - https://www.python.org/dev/peps/pep-0257/#one-line-docstrings. The one you had before was correct also IMO. Your call in the end.

**Pull request merged**